### PR TITLE
Warnint 64to32 6186 v20.1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6186

Describe changes:
- fix `-Wshorten-64-to-32` warnings for some files (c*, f*)

Some commits of #9840
#11257 next batch

### CODE REVIEW

File: src/flow-manager.c [hunk 1]
```diff
@@ -947,7 +947,8 @@ static TmEcode FlowManager(ThreadVars *th_v, void *thread_data)
             gettimeofday(&cond_tv, NULL);
             struct timeval add_tv;
             add_tv.tv_sec = 0;
-            add_tv.tv_usec = (sleep_per_wu * 1000);
+            DEBUG_VALIDATE_BUG_ON(sleep_per_wu > UINT32_MAX / 1000);
+            add_tv.tv_usec = (uint32_t)(sleep_per_wu * 1000);
```
Comment (catenacyber) [inline src/flow-manager.c:951]: Should it be ?
```
 add_tv.tv_sec = sleep_per_wu / 1000;
 add_tv.tv_usec = (sleep_per_wu % 1000) * 1000;
```

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21210

**Comment by victorjulien** [comment]
Can you split off the unrelated commits?

**Comment by catenacyber** [comment]
> Can you split off the unrelated commits?

So, what should I do with these commits ?
They are nit fixes from already merged PRs

**Comment by victorjulien** [comment]
> > Can you split off the unrelated commits?
> 
> So, what should I do with these commits ? They are nit fixes from already merged PRs

Separate PR.

**Comment by catenacyber** [comment]
Continues in https://github.com/OISF/suricata/pull/11416

File: src/flow-manager.c (hunk 1 above)
Comment (victorjulien) [reply src/flow-manager.c:951]: probably yeah

Comment (catenacyber) [reply src/flow-manager.c:951]: ok in next version

**Pull request closed without merging**